src/amo/components/AddonReviewCard/index.js

```diff
@@ -481,7 +486,7 @@ export class AddonReviewCardBase extends React.Component<InternalProps> {
           </a>
         ) : null}
 
-        {flaggable && review ? (
+        {flaggable && review && showFlagButton ? (
```

Comment:
I liked how simple this was in your first patch. It was a lot easier to read. What about this?

```suggestion
        {flaggable && review && (!siteUser || siteUser.id !== review.userId) ? (
```